### PR TITLE
Stop application_off removing paths twice

### PR DESCRIPTION
**Pull request by hsautessella**
The function `application_off` was calling `herbert_off` which removed all paths beneath Herebert's root - except for those in the MATLABPATH environment variable. `application_off` then removed the paths AGAIN, this time also removing the paths in MATLABPATH - which we do not want.

### CODE REVIEW

**Comment by nickbattam-tessella** [comment]
which paths are in MATLABPATH that aren't removed by `herbert_off`?

File: herbert_core/herbert_init.m [hunk 1]
```diff
@@ -127,11 +127,11 @@ function application_off(app_name)
         else
             disp(['Function ',app_name,'_off.m not found in ',rootpath])
             disp('Clearing rootpath and subdirectories from Matlab path in any case')
+            paths = genpath(rootpath);
+            warn_state=warning('off','all');    % turn of warnings (so don't get errors if remove non-existent paths)
+            rmpath(paths);
+            warning(warn_state);    % return warnings to initial state
```
Comment (hsautessella) [inline herbert_core/herbert_init.m]: I think we should still have the logic preventing paths in `MATLABPATH` being removed

**Comment by hsautessella** [comment]
@nickbattam-tessella 
> which paths are in MATLABPATH that aren't removed by `herbert_off`?

Generally `MATLABPATH` is empty, when Matlab starts it adds the user Matlab directory, usually `/home/user/Documents/matlab/` or `~\Documents\Matlab` on Windows.

When we build Hebert `herbert_on.m.template`, `worker4tests.m.template` and `worker_v2.m.template` are copied into the `build/herbert_config` directory (removing the `.template` from the name and updating the path in `herbert_on`). When tests are run through CTest, the `MATLABPATH` environment variable is set to point to `build/herbert_config` so when workers are spawned in MPI tests, they can see herbert_on and the worker files. 

Except, at the moment, when `herbert_init` is called before a test is run, it's removing `Herbert/build/herbert_config` from the path and not re-adding it.

File: herbert_core/herbert_init.m (hunk 1 above)
Comment (abuts) [reply herbert_core/herbert_init.m]: in this form -- yes

Comment (abuts) [reply herbert_core/herbert_init.m]: I do not actually understand why it is removing the MATPABPATH if you have this variable outside of the package. 
You can add this logic to the removal code. 

but in the form I committed, the second removal would not occur so should be irrelevant to MATLABPATH issue

**Comment by abuts** [comment]
you can add test on MATLABPATH settings to the removal script : -- just get its value within the code and add again after removal

**Comment by abuts** [comment]
But as it is only a Jenkins issue, I would rather copy the worker/_on app outside of the Herbert folder.
Unless you are thinking about include setting up this variable into installation/intitialization scripts themselves

**Comment by abuts** [comment]
@hsautessella  This will not work -- new addpath should be moved outside of try/catch block

**Comment by hsautessella** [comment]
> This will not work -- new addpath should be moved outside of try/catch block

Why will it not work? Are we expecting `rmpath` to throw? It works for me

**Comment by abuts** [comment]
If you run herbert_on twice, it will add MATLABPATH to the path first time, and will not do this for the second time
this may cause complex to identify bugs

**Comment by abuts** [comment]
looks fine to me now

**Pull request merged**